test/integration/test-extensions-loading.js

```diff
@@ -21,8 +21,6 @@ function checkElementUpgrade(element) {
   expect(element).to.have.class('-amp-element');
   expect(element).to.have.class('-amp-layout-responsive');
   expect(element).to.have.class('-amp-layout-size-defined');
-  expect(element).to.not.have.class('-amp-notbuilt');
```

Comment:
Yes it passed when I made this change. 

@dvoytenko pointed out that we might be breaking something here so I am doing few other runs with different setup, the safest path he suggested is to bring back the `build` during `discoverWork_` but protect against unready-builds. Testing few combinations now will report back.